hudi-hadoop-mr/src/main/java/org/apache/hudi/hadoop/realtime/RealtimeCompactedRecordReader.java

```diff
@@ -81,7 +82,7 @@ private HoodieMergedLogRecordScanner getMergedLogRecordScanner() throws IOExcept
         .withFileSystem(FSUtils.getFs(split.getPath().toString(), jobConf))
         .withBasePath(split.getBasePath())
         .withLogFilePaths(split.getDeltaLogPaths())
-        .withReaderSchema(usesCustomPayload ? getWriterSchema() : getReaderSchema())
+        .withReaderSchema(getWriterSchema())
```

Comment:
why this change? here, we're saying that let's set the reader schema to be the writer schema, but then let's say `usesCustomPayload` is false and `supportPayload` is also false (due to fallback) then `buildGenericRecordwithCustomPayload` will actually want to use reader schema while calling `getInsertValue`. In case of mismatch between reader and writer schema, won't there be an error?
Should we revert this and set the schema based on conditional as it was before like below?
`.withReaderSchema(usesCustomPayload ? getWriterSchema() : getReaderSchema())`

Reply:
yes， agree.  update the codes